src/ray/raylet/scheduling/cluster_resource_manager.cc

```diff
@@ -292,4 +292,22 @@ BundleLocationIndex &ClusterResourceManager::GetBundleLocationIndex() {
   return bundle_location_index_;
 }
 
+void ClusterResourceManager::UpdateNodeLabels(
+    const scheduling::NodeID &node_id,
+    const absl::flat_hash_map<std::string, std::string> &labels) {
+  if (labels.empty()) {
+    return;
+  }
+
+  auto it = nodes_.find(node_id);
+  if (it == nodes_.end()) {
+    NodeResources node_resources;
+    it = nodes_.emplace(node_id, node_resources).first;
+  }
```

Comment:
it's a bit weird that we can have node with only labels but empty resources. We should be able to set labels and resources together when we add a node?

Reply:
I'm actually wondering when we may not have total resources. Should it be a check instead of warning? Let me check with the team.

Reply:
Can we actually change `if (it == nodes_.end()) {` to CHECK which makes code easier to reason about.

Reply:
My current implementation requires me to retain this logic. Since the labels information is already available in 'GcsNodeInfo', I will first call 'ResetNodeLabels()', and then update the resources.
![image](https://github.com/ray-project/ray/assets/11072802/140c4f0f-86f7-43c8-acc7-e60e2151039d)